ocs_ci/deployment/rhv.py

```diff
@@ -0,0 +1,107 @@
+"""
+This module contains platform specific methods and classes for deployment
+on Red Hat Virtualization (RHV) platform
+"""
+import logging
+import os
+
+from ocs_ci.deployment.ocp import OCPDeployment as BaseOCPDeployment
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.utility.bootstrap import gather_bootstrap
+from ocs_ci.utility.utils import run_cmd
+from .deployment import Deployment
+
+logger = logging.getLogger(__name__)
+
+# As of now only IPI
+__all__ = ["RHVIPI"]
+
+
+class RHVBASE(Deployment):
+
+    # default storage class for StorageCluster CRD on RHV platform
+    DEFAULT_STORAGECLASS = "ovirt-csi-sc"
+
+    def __init__(self):
+        """
+        This would be base for both IPI and UPI deployment
+        """
+        super(RHVBASE, self).__init__()
+
+        if config.ENV_DATA.get("default_cluster_name"):
+            config.ENV_DATA["cluster_name"] = config.ENV_DATA["default_cluster_name"]
+
+
+class RHVIPI(RHVBASE):
+    """
+    A class to handle RHV IPI specific deployment
+    """
+
+    def __init__(self):
+        super(RHVIPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super(RHVIPI.OCPDeployment, self).__init__()
+
+        def deploy_prereq(self):
+            """
+            Overriding deploy_prereq from parent. Perform all necessary
+            prerequisites for RHVIPI here.
+            """
+            super(RHVIPI.OCPDeployment, self).deploy_prereq()
+            if config.DEPLOYMENT["preserve_bootstrap_node"]:
+                logger.info("Setting ENV VAR to preserve bootstrap node")
+                os.environ["OPENSHIFT_INSTALL_PRESERVE_BOOTSTRAP"] = "True"
+                assert os.getenv("OPENSHIFT_INSTALL_PRESERVE_BOOTSTRAP") == "True"
+
+        def deploy(self, log_cli_level="DEBUG"):
```

Comment:
this `deploy()` method is similar to the one implemented in `ocs_ci.deployment.vmware.VSPHEREIPI.OCPDeployment.deploy`.

For cloud based platforms, we have `ocs_ci/deployment/cloud.py`. How about we do the same for on-prem platforms and move such similar class methods there?

Reply:
Yes, this sounds good.